File: Python3/_167.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        low, high = 0, len(nums) - 1
        
        while (low < high):
            sum_ = nums[low] + nums[high]
            
            if sum_ == target:
            	# indices given as range 1 to n (1-indexed)
                return [low + 1, high + 1]
                
            # Case: Invalidate nums[high] as candidate because
            # its sum with lowest still-viable candidate still 
            # exceeds target
            elif sum_ > target:
                high -= 1
            else:
                low += 1
    	

File: Cpp/_167.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> nums, int target) {
        int low = 0, high = nums.size() - 1;

        while (low < high) {
            int sum = nums[low] + nums[high];

            if (sum == target)
                return vector<int> {low + 1, high + 1};
            else if (sum > target)
                high--;
            else
                low++;
        }
        
        return vector<int> {-1, -1};
    }
};